arbor/backends/multicore/cable_solver.hpp

```diff
@@ -74,71 +66,82 @@ struct cable_solver {
         }
     }
 
-    const_view solution() const {
-        // In this back end the solution is a simple view of the rhs, which
-        // contains the solution after the matrix_solve is performed.
-        return rhs;
-    }
+    // Setup and solve the cable equation
+    // * expects the voltage from its first argument
+    // * will likewise overwrite the first argument with the soluction
+    template<typename T>
+    void solve(T& rhs, const_view dt_intdom, const_view current, const_view conductivity) {
+        value_type * const __restrict__ d_ = d.data();
```

Comment:
`__restrict__` is not a c++ keyword, it is implemented by gcc and clang, though. For visual c++ it would be `__declspec(restrict)`. Maybe we should put the `restrict` keyword behind an arbor macro à la `ARB_RESTRICT`?

Reply:
NVCC and XLC support it as well; at least. We do not target MSVC. 
That said, it's a good idea regardless. It also plan to employ it in the ABI
soon.